dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerTest.kt

```diff
@@ -0,0 +1,177 @@
+package org.dataland.e2etests.tests
+
+import org.dataland.datalandbackend.openApiClient.api.CompanyDataControllerApi
+import org.dataland.datalandbackend.openApiClient.api.EuTaxonomyDataControllerApi
+import org.dataland.datalandbackend.openApiClient.api.MetaDataControllerApi
+import org.dataland.datalandbackend.openApiClient.infrastructure.ClientException
+import org.dataland.datalandbackend.openApiClient.model.CompanyAssociatedDataEuTaxonomyData
+import org.dataland.datalandbackend.openApiClient.model.DataMetaInformation
+import org.dataland.datalandbackend.openApiClient.model.StoredCompany
+import org.dataland.e2etests.BASE_PATH_TO_DATALAND_PROXY
+import org.dataland.e2etests.TestDataProvider
+import org.dataland.e2etests.accessmanagement.TokenRequester
+import org.dataland.e2etests.accessmanagement.UnauthorizedCompanyDataControllerApi
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class CompanyDataControllerTest {
+
+    private val metaDataControllerApi = MetaDataControllerApi(BASE_PATH_TO_DATALAND_PROXY)
+    private val companyDataControllerApi = CompanyDataControllerApi(BASE_PATH_TO_DATALAND_PROXY)
+    private val euTaxonomyDataControllerApi = EuTaxonomyDataControllerApi(BASE_PATH_TO_DATALAND_PROXY)
+    private val testDataProvider = TestDataProvider()
+    private val tokenRequester = TokenRequester()
+    private val unauthorizedCompanyDataControllerApi = UnauthorizedCompanyDataControllerApi()
+
+    private fun postOneCompanyAndData(): Map<String, String> {
+        tokenRequester.requestTokenForUserType(TokenRequester.UserType.Admin).setToken()
+        val testCompanyInformation = testDataProvider.getCompanyInformation(1).first()
+        val testData = testDataProvider.getEuTaxonomyData(1).first()
+        val testDataType = testData.javaClass.kotlin.qualifiedName!!.substringAfterLast(".")
+        val testCompanyId = companyDataControllerApi.postCompany(testCompanyInformation).companyId
+        val testDataId = euTaxonomyDataControllerApi.postCompanyAssociatedData(
+            CompanyAssociatedDataEuTaxonomyData(testCompanyId, testData)
+        ).dataId
+        return mapOf("companyId" to testCompanyId, "dataId" to testDataId, "dataType" to testDataType)
+    }
+
+    @Test
+    fun `post a dummy company and check if post was successful`() {
+        val testCompanyInformation = testDataProvider.getCompanyInformation(1).first()
+        tokenRequester.requestTokenForUserType(TokenRequester.UserType.Admin).setToken()
```

Comment:
Der Aufruf für diesen Tokenrequester ist immer sehr ähnlich. Der Token selbst wird immer nur als Input für setToken verwendet. Könnte man das nicht gleich in der Klasse abhandeln. Dann könnte man vermutlich auch die Token Klasse da mit rein nehmen da diese ohnehin nicht anderweitig verwendet wird.

Reply:
Ja wäre kein Problem.

Ich hatte die Token-Klasse "getrennt" geschrieben, weil wir sonst immer eine Klasse pro File haben im Repo.
Wir haben z.B. für einzeilige Enum-Klassen in den models jeweils ein .kt-File erstellt.
Ich hatte deswegen implizit angenommen, dass das eine Art Design-Vorgabe/Regel für das Projekt ist, und deswegen die Token-Klasse "getrennt" erstellt.

Wenn wir aber keine solche Regeln verfolgen, fände ich es auch besser so wie du es beschreibst.